golang/pkg/crane/k8s/service.go

```diff
@@ -125,7 +125,7 @@ func getServicePorts(portBindings []builder.PortBinding, portRanges []builder.Po
 	ports := []*acorev1.ServicePortApplyConfiguration{}
 
 	for i := range portBindings {
-		portNum := int32(portBindings[i].PortBinding)
+		portNum := int32(*portBindings[i].PortBinding)
```

Comment:
Does  `PortBinding` always have a value in this case?

Reply:
Added fallback: using internal for external if external is not defined.